lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -357,11 +357,15 @@ static void checkInheritanceClause(
       }
     }
 
+    SourceRange removeRange = inherited.getSourceRange();
+
     // We can't inherit from a non-class, non-protocol type.
     decl->diagnose(canHaveSuperclass
-                   ? diag::inheritance_from_non_protocol_or_class
-                   : diag::inheritance_from_non_protocol,
-                   inheritedTy);
+                       ? diag::inheritance_from_non_protocol_or_class
+                       : diag::inheritance_from_non_protocol,
+                   inheritedTy)
+        .fixItRemoveChars(removeRange.Start, removeRange.End);
```

Comment:
Nit: I think there is a function that takes source range already `.fixItRemove(inherited.getSourceRange());`.